$MirOS: ports/net/rrdtool/patches/patch-src_rrd_dump_c,v 1.1 2008/11/12 20:27:50 tg Exp $
--- src/rrd_dump.c.orig	Mon Apr 25 21:03:22 2005
+++ src/rrd_dump.c	Wed Nov 12 22:32:43 2008
@@ -39,8 +39,8 @@ rrd_dump(int argc, char **argv)    
 #else
 # error "Need strftime"
 #endif
-    printf("\t<lastupdate> %ld </lastupdate> <!-- %s -->\n\n",
-	   rrd.live_head->last_up,somestring);
+    printf("\t<lastupdate> %lld </lastupdate> <!-- %s -->\n\n",
+	   (int64_t)rrd.live_head->last_up,somestring);
     for(i=0;i<rrd.stat_head->ds_cnt;i++){
 	printf("\t<ds>\n");
 	printf("\t\t<name> %s </name>\n",rrd.ds_def[i].ds_nam);
@@ -126,7 +126,7 @@ rrd_dump(int argc, char **argv)    
 #else
 # error "Need strftime"
 #endif
-	    printf("\t\t\t<!-- %s / %d --> <row>",somestring,(int)now);
+	    printf("\t\t\t<!-- %s / %lld --> <row>",somestring,(int64_t)now);
 	    for(iii=0;iii<rrd.stat_head->ds_cnt;iii++){			 
 		fread(&my_cdp,sizeof(rrd_value_t),1,in_file);		
 		if (isnan(my_cdp)){
